File: backend/src/main/java/ch/rodano/core/services/bll/configurator/ScopeModelRightsService.java
package ch.rodano.core.services.bll.configurator;

import java.util.Map;
import java.util.UUID;

import ch.rodano.api.config.EntityRightDTO;

public interface ScopeModelRightsService {

	Map<UUID, Map<UUID, EntityRightDTO>> getScopeModelRights(UUID projectId);

	void saveScopeModelRights(UUID projectId, Map<UUID, Map<UUID, EntityRightDTO>> rights);
}

File: frontends/main/src/app/configurator/services/api/scope-model-rights.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {EntityRight} from '@core/model/entity-right';

@Injectable({providedIn: 'root'})
export class ScopeModelRightsService {
	constructor(private http: HttpClient) {}

	getRights(projectId: string): Observable<Record<string, Record<string, EntityRight>>> {
		return this.http.get<Record<string, Record<string, EntityRight>>>(
			`/api/superuser/configurator/projects/${projectId}/config/scope-model-rights`
		);
	}

	saveRights(projectId: string, rights: Record<string, Record<string, EntityRight>>): Observable<void> {
		return this.http.put<void>(
			`/api/superuser/configurator/projects/${projectId}/config/scope-model-rights`,
			rights
		);
	}
}
